src/main/java/org/alfresco/heartbeat/HBDataCollectorServiceImpl.java

```diff
@@ -109,4 +276,68 @@ public synchronized void enabled(boolean enabled)
         this.hbDataSenderService.enable(enabled);
     }
 
+    /**
+     * The scheduler job responsible for triggering a heartbeat on a regular basis.
+     */
+    public class HeartBeatJob implements Job
+    {
+        public void execute(final JobExecutionContext jobexecutioncontext) throws JobExecutionException
+        {
+            String jobName = jobexecutioncontext.getJobDetail().getName();
+            QName qName = QName.createQName(NamespaceService.SYSTEM_MODEL_1_0_URI, jobName);
+            String lockToken = null;
+            try
+            {
+                // Get a lock
+                lockToken = jobLockService.getLock(qName, LOCK_TTL);
```

Comment:
If you put a TTL of 30s on the job then you need to be sure that all jobs finish in that time; this cannot be guaranteed.
That is why you should either
- refresh the lock at regular intervals
- use the job lock callback pattern (plenty of examples) and an AtomicBoolean to keep track of whether the job is still active or not.